### ISSUE TITLE
TEXMFHOME not searched by texstudio after homebrew texlive install (after switching to Ventura?)

### ISSUE THREAD

**Issue by gpasa** [texstudio-org/texstudio#2974]
I was previously on texlive under MacOS Macports, and using texstudio without any problem. I have a bunch of custom packages in ~/Library/texmf and when I build my documents on the command line in terminal everything is fine. I have migrated to homebrew and moved away from MacPorts. Everything works fine from the terminal, but I cannot build my documents from texstudio anymore, as the packages in ~/Library/texmf are not found, although they are found when building from the cli. Does anybody know what the problem is ?

I tried to texhash, but in the terminal it does not check TEXMFHOME (which I set in .profile). Everything seems fine, but from texstudio pdflatex does not find the packages in my home folder ~/Library/texmf. This drives me mad!

I already tried to "Try to set the texmf folder with tlmgr conf texmf TEXMFHOME ~/Library/texmf/" (had to add conf to tlmgr/config first), but still gets ! LaTeX Error: File `mymath.sty' not found.

From the Terminal everything is fine. TEXMFHOME is fine too, and the commands run flawlessly. The .profile is correctly set up. It's only from Texstudio that ~/Library/texmf isn't searched for. For now, I got it work by launching texstudion from the command line in a Terminal. I think that (as samcarter_is_at_topanswers.xyz said, that GUI apps do not have the correct environment for TEXMFHOME, or something related). BTW: I got this problem after upgrading (sic! ) to Ventura (maybe it's related...) – 

I could check that if I start the app by double-clicking its icon or from the Spotlight, there is a problem with the search path. But if I lauch it by double-clicking the executable (after Dispalying the content of the package, and clicking on the executable in MAcOS folder) then the path is initialized correctly and everything is fine. Seems like opening the app as usual does not set the paths correctly. – 

I could figure out that texstudio started from Spotlight or by double-clicking the icon, does not have its environment variables set properly (I think it does not read the environment variables in the same way as when started from a terminal window). In particular TEXMFHOME is not initialized from direct icon launch. But everything is ok from the command line. How can I correct this behavior?

**Comment by sunderme** [comment]
are you sure that there is only *one* latex installation and the same pdflatex is used ?

**Comment by gpasa** [comment]
yes. I simply started a user command "echo $TEXMFHOME", and when launched from texstudio icon launched it displays nothing, but if launched on texstudio started from the cli, then it displays the TEXMFHOME folder.

**Comment by muzimuzhi** [comment]
`TEXMFHOME` works fine for me, with macOS 12.6.2 + texstudio 4.5.1 (opened from Spotlight).  For example, `\usepackage{abc}` loads `/Users/<username>/Library/texmf/tex/latex/abc.sty`.

> I simply started a user command "echo $TEXMFHOME", and when launched from texstudio icon launched it displays nothing, but if launched on texstudio started from the cli, then it displays the TEXMFHOME folder.

Yes I think by default texstudio won't have access to shell environment variables. But you also don't have to set a env var called `TEXMFHOME`. By default the tlmgr variable `TEXMFHOME` is used, whose value is written in a `texmf.cnf` file
```
$ tlmgr conf texmf TEXMFHOME
tlmgr: texmf TEXMFHOME value: ~/Library/texmf (/usr/local/texlive/2022basic/texmf.cnf)
# an alternative way
$ kpsewhich --var-value TEXMFHOME
/Users/guten/Library/texmf
```

Since it seems setting `tlmgr conf texmf TEXMFHOME <path>` didn't work for you, sorry I don't know what and where to check next.

As a workaround, you can modify texstudio's build commands to explicitly launch a shell, it can set per tex doc as a [magic comment](https://texstudio-org.github.io/advanced.html#advanced-header-usage), or set globally in the TeXstudio Configuration - Commands (or Build).
```
% !TeX program = sh -c 'export TEXMFHOME=~/Library/texmf; pdflatex -synctex=1 -interaction=nonstopmode %.tex'
```

**Comment by sunderme** [comment]
A bit of googling suggests:
`launchctl setenv TEXMFHOME ...`
Maybe someone can figure out, how to make that permanent ?

**Comment by gpasa** [comment]
Finaly the modification of texstudio launch commands works if I set
 sh -c 'TEXMFHOME="~/Library/texmf" pdflatex -synctex=1 -interaction=nonstopmode %.tex'
 
 Haven't tried your last suggestion.

**Comment by muzimuzhi** [comment]
@gpasa What do you get in the log panel, Messages tab by running the example below? You can use texstudio's output redirection feature to test more, see it's doc in https://texstudio-org.github.io/configuration.html#output-redirection
```tex
% !TeX TS-program = kpsewhich --var-value TEXMFHOME > txs:///messages | pdflatex -interact=nonstopmode %.tex
\documentclass{article}
\begin{document}
content
\end{document}
```
Here's what I got (consecutive empty lines removed)
```
Process started: kpsewhich --var-value TEXMFHOME

/Users/guten/Library/texmf
Process exited normally

Process started: pdflatex -interact=nonstopmode "texstudio_VciKGQ".tex

Process exited normally
```

**Comment by sunderme** [comment]
so, can we consider this as solved ?

**Comment by gpasa** [comment]
If putting the command 
sh -c 'TEXMFHOME="~/Library/texmf" pdflatex -synctex=1 -interaction=nonstopmode %.tex'

for pdflatex and similar commands for the other application is the solution, then I think that the issue is closed.

**Comment by muzimuzhi** [comment]
Setting env var is sort of a should-always-work workaround. The real cause is still uncovered.
